### ISSUE TITLE
Bootstrap fails on GNU/Linux (WSL)

### ISSUE THREAD

**Issue by HauserV** [Microsoft/vcpkg#4314]
*The machine setup:* I'm using Debian through WSL (4.4.0-17134-Microsoft) on top of Windows 10 Education v1803, build 17134.285. I am using GCC 8.2.0-6 and CMake 3.12.1.
*The steps to reproduce:* I clone the vcpkg repository and run `./bootstrap-vcpkg.sh`
*The outcome you expected:* I expected the build to successfully complete.
*The actual outcome:* I get the following error caused by the fact that [`vcpkg::VisualStudio` declaration](https://github.com/Microsoft/vcpkg/blob/a1a18eada148562d0906b103c70c9e36b6a648ce/toolsrc/include/vcpkg/visualstudio.h#L7) is guarded by `#if defined(_WIN32)` and therefore is *not* visible to the compiler.

```
[32/64] Building CXX object CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o
FAILED: CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o
/usr/bin/g++-8  -DDISABLE_METRICS=0 -I../include -O3 -DNDEBUG   -std=c++1z -pthread -MD -MT CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o -MF CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o.d -o CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o -c ../src/vcpkg/commands.xvsinstances.cpp
../src/vcpkg/commands.xvsinstances.cpp: In function ‘void vcpkg::Commands::X_VSInstances::perform_and_exit(const vcpkg::VcpkgCmdArguments&, const vcpkg::VcpkgPaths&)’:
../src/vcpkg/commands.xvsinstances.cpp:21:39: error: ‘vcpkg::VisualStudio’ has not been declared
         const auto instances = vcpkg::VisualStudio::get_visual_studio_instances(paths);
                                       ^~~~~~~~~~~~
../src/vcpkg/commands.xvsinstances.cpp:22:44: error: unable to deduce ‘auto&&’ from ‘instances’
         for (const std::string& instance : instances)
                                            ^~~~~~~~~
```

**Comment by LeonineKing1199** [comment]
Can confirm this bug as well. I'm OSX, using gcc-8 as well.

```
[33/64] Building CXX object CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o
FAILED: CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o 
/usr/local/bin/g++-8  -DDISABLE_METRICS=0 -I../include -O3 -DNDEBUG   -std=c++1z -MD -MT CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o -MF CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o.d -o CMakeFiles/vcpkg.dir/src/vcpkg/commands.xvsinstances.cpp.o -c ../src/vcpkg/commands.xvsinstances.cpp
../src/vcpkg/commands.xvsinstances.cpp: In function 'void vcpkg::Commands::X_VSInstances::perform_and_exit(const vcpkg::VcpkgCmdArguments&, const vcpkg::VcpkgPaths&)':
../src/vcpkg/commands.xvsinstances.cpp:21:39: error: 'vcpkg::VisualStudio' has not been declared
         const auto instances = vcpkg::VisualStudio::get_visual_studio_instances(paths);
                                       ^~~~~~~~~~~~
../src/vcpkg/commands.xvsinstances.cpp:22:44: error: unable to deduce 'auto&&' from 'instances'
         for (const std::string& instance : instances)
                                            ^~~~~~~~~
```

**Comment by ras0219-msft** [comment]
This was a regression that accidentally made its way into master -- we've committed a fix at 1faf5c7d289f5b693fbcaa98af8b41833e6a5a04 which should have solved this issue!

Please `git pull` and re-run bootstrap.

Sorry for the disruption and for future incidents they can be worked around by temporarily checking out a previous commit that worked for you (`git reflog` can be helpful here)!
